qiskit/transpiler/passes/synthesis/linear_functions_synthesis.py

```diff
@@ -13,30 +13,25 @@
 
 """Synthesize LinearFunctions."""
 
-from qiskit.converters import circuit_to_dag
 from qiskit.transpiler.basepasses import TransformationPass
 from qiskit.dagcircuit.dagcircuit import DAGCircuit
 from qiskit.circuit.library import Permutation
 from qiskit.circuit.exceptions import CircuitError
+from qiskit.transpiler.passes.synthesis.high_level_synthesis import HighLevelSynthesis, HLSConfig
 
 
-class LinearFunctionsSynthesis(TransformationPass):
-    """Synthesize linear functions. Under the hood, this runs cnot_synth
-    which implements the Patel–Markov–Hayes algorithm."""
+class LinearFunctionsSynthesis(HighLevelSynthesis):
```

Comment:
I think I missed / didn't correctly understand all the actual discussion of this in the last two Terra meetings, but it seems unusual to me that this would derive from the plugin _runner_, and not be a plugin itself (i.e. derive `HighLevelSynthesisPlugin`) that registers itself as running on `linear_function` names.

Reply:
The discussion on `LinearFunctionsSynthesis` is something that I've mentioned only in the very last Terra meeting. To try to explain this better, `LinearFunctionsSynthesis` is a transpiler pass (that was introduced when we introduced `LinearFunctions`) that walks over a DAG and replaces `LinearFunction` objects by their definitions. We no longer need this pass after introducing a much more general `HighLevelSynthesis` transpiler pass, as
`LinearFunctionSynthesis().run(dag)` is currently completely equivalent to `HighLevelSynthesis(hls_config=default_linear_config).run(dag)`, where `default_linear_config = HLSConfig(linear_function=[[("default", {})]], use_default_on_unspecified=False)`. I think that we have agreed that it's ok to deprecate this pass (I will add the deprecation warning code shortly), but meanwhile it makes sense to make `LinearFunctionSynthesis` to be consistent with `HighLevelSynthesis`.

Reply:
Adding deprecation warnings in [e87e952](https://github.com/Qiskit/qiskit-terra/pull/9041/commits/e87e9528e4d74de12c11909cbfaa7fdf40ed3f78).